### PR TITLE
Remove constraint of using nullable unions

### PR DESCRIPTION
**Pull request by iperevozchikov**
Issue: #412

### CODE REVIEW

File: projects/schema-form/src/lib/model/nullproperty.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import {ISchema} from './ISchema';
+import { LogService } from '../log.service';
+
+export class NullProperty extends AtomicProperty {
+  fallbackValue() {
+    return null;
+  }
+}
+
+PROPERTY_TYPE_MAPPING.boolean = (
```
Comment (anotherpit) [inline projects/schema-form/src/lib/model/nullproperty.ts:15]: `boolean` name is already in use by `BooleanProperty` model. you meant `PROPERTY_TYPE_MAPPING.null = ...`, perhaps

File: projects/schema-form/src/lib/model/ISchema.ts [hunk 2]
```diff
@@ -37,7 +38,20 @@ export interface IProperties {
...
+
+export type TSchemaNullableScalarPrimitiveType = TUnorderedPair<
+  TSchemaNullPrimitiveType,
+  TSchemaScalarPrimitiveType
+>;
+
+export type TSchemaPropertyType = FieldType
+  | TNullableFieldType
+  | TSchemaPrimitiveType
+  | TSchemaNullableScalarPrimitiveType;
```
Comment (anotherpit) [inline projects/schema-form/src/lib/model/ISchema.ts:54]: hm? isn't `FieldType | TNullableFieldType` enough? `TSchemaPrimitiveType` is just a copy of `FieldType` enum values? why do we need both enum and strings at all?

Comment (iperevozchikov) [reply projects/schema-form/src/lib/model/ISchema.ts:54]: Library has another variant of working with json-schema over template syntax, and there uses FieldType enum. I don't change the logic inside this type, just a little bit cleaned up

File: projects/schema-form/src/lib/model/nullproperty.ts (hunk 1 above)
Comment (iperevozchikov) [reply projects/schema-form/src/lib/model/nullproperty.ts:15]: yep, it's mistake. will change on null key

Comment (iperevozchikov) [reply projects/schema-form/src/lib/model/nullproperty.ts]: :white_check_mark:  fixed

**Review by ebrehault — APPROVED** [review]

**Pull request merged**